packages/starlight/utils/i18n.ts

```diff
@@ -161,6 +169,23 @@ function getLocaleInfo(lang: string) {
 	}
 }
 
+/**
+ * Returns the direction of the passed locale.
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getTextInfo
+ */
+function getLocaleDir(locale: Intl.Locale): 'ltr' | 'rtl' {
+	if ('textInfo' in locale) {
+		// @ts-expect-error - `textInfo` is typed but is available in v8 based environments.
+		return locale.textInfo.direction;
+	} else if ('getTextInfo' in locale) {
+		// @ts-expect-error - `getTextInfo` is typed but is available in some non-v8 based environments.
+		return locale.getTextInfo().direction;
+	}
+	// Firefox does not support `textInfo` or `getTextInfo` yet so we fallback to a well-known list
+	// of right-to-left languages.
+	return wellKnownRTL.includes(locale.language) ? 'rtl' : 'ltr';
```

Comment:
Is `locale.language` already stripped of any regional identifier or could this also be `ar-SA` in theory?

Reply:
Indeed, this would already be stripped, e.g. here is an example in Firefox (as I would assume this would be the only browser hitting this path):

<img width="253" alt="image" src="https://github.com/withastro/starlight/assets/494699/7691f190-1741-4338-82da-e130fdce8a1c">